clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2990,9 +2991,23 @@ static void setLinkageForGV(llvm::GlobalValue *GV, const NamedDecl *ND) {
     GV->setLinkage(llvm::GlobalValue::ExternalWeakLinkage);
 }
 
+static bool hasExistingGeneralizedTypeMD(llvm::Function *F) {
+  llvm::MDNode *MD = F->getMetadata(llvm::LLVMContext::MD_type);
+  if (!MD)
+    return false;
+  return hasGeneralizedMDString(MD);
```

Comment:
I thought this function did checking for nullptr? if so, do we need the helper at all?